### PR TITLE
add segment_min to tensorflow frontend and test

### LINKED ISSUE #23079: segment_min

### PR DESCRIPTION
**Pull request by chihangs**
# add segment_min to tensorflow frontend and test

## sub task of: Add Math functions for TensorFlow frontend #1545

Close #23079

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 1]
```diff
@@ -2522,6 +2522,39 @@ def test_tensorflow_scalar_mul(
...
 
+# segment_min
+@handle_frontend_test(
+    fn_tree="tensorflow.math.segment_min",
+    data=helpers.array_values(
+        dtype=ivy.float64, shape=(5, 6), min_value=1, max_value=9
+    ),
+    segment_ids=helpers.array_values(
+        dtype=ivy.int64, shape=(5,), min_value=0, max_value=4
+    ),
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:2533]: Can we change these to use `dtype_and_x` like the other tests?

File: ivy/functional/frontends/tensorflow/math.py [hunk 2]
```diff
@@ -605,6 +605,24 @@ def scalar_mul(scalar, x, name="scalar_mul"):
     return ivy.multiply(x, scalar).astype(x.dtype)
 
 
+@to_ivy_arrays_and_back
+def segment_min(data, segment_ids, name="segment_min"):
+    if ivy.min(segment_ids) < 0:
+        raise ivy.utils.exceptions.IvyException("segments_ids must be non-negative")
+    # if ivy.as_ivy_dev(ivy.dev(segment_ids)).find('cpu') > -1:
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/tensorflow/math.py:612]:
```suggestion
```

File: ivy/functional/frontends/tensorflow/math.py [hunk 3]
```diff
@@ -605,6 +605,24 @@ def scalar_mul(scalar, x, name="scalar_mul"):
     return ivy.multiply(x, scalar).astype(x.dtype)
 
 
+@to_ivy_arrays_and_back
+def segment_min(data, segment_ids, name="segment_min"):
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/tensorflow/math.py:609]:
```suggestion
def segment_min(data, segment_ids, name=None):
```

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@chihangs I've left some comments 😊 please could you also remove the change the test_array_api? 

Is the test you wrote passing for you? It seems to be running extremely slowly for me, maybe refactoring it to be close to the other tests will fix this, or it could be that your frontend implementation is incorrect.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py (hunk 1 above)
Comment (chihangs) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py]: done

**Comment by chihangs** [comment]
> @chihangs I've left some comments 😊 please could you also remove the change the test_array_api?
> 
> Is the test you wrote passing for you? It seems to be running extremely slowly for me, maybe refactoring it to be close to the other tests will fix this, or it could be that your frontend implementation is incorrect.

On speed: I guess the reason is that each of most other math functions only has one matrix or even scalar input.
On  test_array_api - not sure why a bunch of files being flagged as changed. As I don't have write access of unify/ivy, can you only accept changes of other files and not this folder?

**Comment by Sam-Armstrong** [comment]
@chihangs please could you merge the latest changes to ivy into this PR? I think you will have to update your fork. Also your test is failing for me with an error which is something like this: `TypeError: Field elements must be 2- or 3-tuples, got ''int8''`

**Comment by chihangs** [comment]
@Sam-Armstrong  if I understand correctly,  we only need to look at the display_test_results job and lint / Check formatting / check-formatting tests, right? At the bottom of the last check, I can see both are passed successfully. So would this imply that you can accept the code now?

**Comment by Sam-Armstrong** [comment]
@chihangs You're better off running the test you wrote yourself to see what's going on. You can do this with the command `pytest ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py::test_tensorflow_segment_min` from the terminal. On my machine the test you wrote is failing for all backends with errors similar to `TypeError: Field elements must be 2- or 3-tuples, got ''int64''`. Could you run them and see if you get the same?

**Comment by chihangs** [comment]
> @chihangs You're better off running the test you wrote yourself to see what's going on. You can do this with the command `pytest ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py::test_tensorflow_segment_min` from the terminal. On my machine the test you wrote is failing for all backends with errors similar to `TypeError: Field elements must be 2- or 3-tuples, got ''int64''`. Could you run them and see if you get the same?

@Sam-Armstrong Thanks very much for your prompt reply! It seems to have connection error to your google cloud Redis server. Would you confirm whether I have access right to this?

**Comment by Sam-Armstrong** [comment]
@chihangs please could you give me some more information about the error(s) your seeing? I'm not really sure what you mean about google cloud Redis server, shouldn't the test be able to run locally?

**Comment by chihangs** [comment]
> @chihangs please could you give me some more information about the error(s) your seeing? I'm not really sure what you mean about google cloud Redis server, shouldn't the test be able to run locally?

@Sam-Armstrong it shows sth like this: 
redis.exceptions.ConnectionError: Error while reading from redis-17011.c259.us-central1-2.gce.cloud.redislabs.com:17011 : (54, 'Connection reset by peer') .....
redis.exceptions.ConnectionError: Error 8 connecting to redis-17011.c259.us-central1-2.gce.cloud.redislabs.com:17011. nodename nor servname provided, or not known.

**Comment by Sam-Armstrong** [comment]
Are you running the exact command I sent? Are you running it locally through docker?

**Comment by chihangs** [comment]
> @chihangs Are you running the exact command I sent? Are you running it locally through docker?

@Sam-Armstrong yes exactly same command, by conda virtual environment in Mac terminal

**Comment by Sam-Armstrong** [comment]
@chihangs hmm I'm not really sure what the problem is. Maybe you could try setting up the environment with docker? https://unify.ai/docs/ivy/overview/contributing/setting_up.html#docker-interpreter-with-pycharm

**Comment by chihangs** [comment]
@Sam-Armstrong suppose your testing helper function is testing by comparing with backend function? But the backend segment_min isn't implemented, so this should not be my fault? On docker, your documentation says it requires PyCharm paid/Professional version, which is not available at my end.

**Comment by Sam-Armstrong** [comment]
have messaged you on discord :)

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**